scvi/external/cellassign/_model.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+import pandas as pd
+import torch
+from anndata import AnnData
+
+from scvi.dataloaders import ScviDataLoader
+from scvi.external.cellassign._module import CellAssignModule
+from scvi.lightning import VAETask
+from scvi.model.base import BaseModelClass
+
+
+class CellAssign(BaseModelClass):
+    """
+    Reimplementation of CellAssign for reference-based annotation.
+
+    Parameters
+    ----------
+    adata
+        single-cell AnnData object that has been registered via :func:`~scvi.data.setup_anndata`.
+    cell_type_markers
+        Binary marker gene matrix
+    use_gpu
+        Use the GPU or not.
+    **model_kwargs
+        Keyword args for :class:`~scvi.external.CellAssignModels`
+
+    Examples
+    --------
+    >>> # TODO
+    """
+
+    def __init__(
+        self,
+        sc_adata: AnnData,
+        cell_type_markers: pd.DataFrame,
```

Comment:
this will be a binary matrix, genes by cell types (rho)

`sc_adata` --> fancy pandas dataframe, gene names are in `sc_adata.var.index`

check that genes are the same in both objects, reorder `cell_type_markers` according to order of genes in `sc_adata.var.index`